core/group-sign_ZZZ.c.in

```diff
@@ -808,7 +781,7 @@ static int join_finish_client(struct GroupPublicKey *pub, struct UserPrivateKey
         return 0;
     }
 
-    if (!verifyAux(&resp->cred.A, &resp->cred.B, &resp->cred.C, &resp->cred.D, &pub->X, &pub->Y)) {
+    if (!verifyAuxFast(&resp->cred.A, &resp->cred.B, &resp->cred.C, &resp->cred.D, &pub->X, &pub->Y, RNG)) {
```

Comment:
Do you remember why we had different verify implementations on client and server?

Reply:
Basically because you have to introduce this random number generator for the probabilistic fast verification, and since join performance in client was not that big issue I left it as is. But with this trick (seeding with the user secret key) not so much code needs to be changed.